### ISSUE TITLE
inconsitency between character changeview and speechview

### ISSUE THREAD

**Issue by selmiak** [adventuregamestudio/ags#729]
it is
cCharacter.Changeview(#);
but
cCharacter.Speechview = #;

I searched the manual for ChangeSpeechView and didn't find it. It could also be cChar.View =#; or cChar.ChangeSpeechView(#);
or accept both.
just a suggestion for consitency in scripting.

**Comment by ivan-mogilko** [comment]
I'd like to point out that there is already a property for ChangeView: it is **Character.NormalView**.
Except it is *readonly*.

From the manual:

`readonly int Character.NormalView`

> Gets the character's normal view. This is the character's standard walking view, that is used when his view is not locked to something else.
This property is read-only; to change it, use the ChangeView command.

The Character.View property, on the other hand, tells *current* view, which may be either NormalView, or locked view (set with Character.LockView).

So, IMHO, question is, why is Character.NormalView readonly and cannot it be made writeable to replace ChangeView completely?

**Comment by ericoporto** [comment]
I found a comment from CJ explaining the choice of API: https://www.adventuregamestudio.co.uk/forums/advanced-technical-forum/ags-v2-7-final-yet-another-edition/msg227721/#msg227721

**Comment by ivan-mogilko** [comment]
My question was not why you cannot modify a View property, but why is not there a property for "normal walking view", which is set using ChangeView. I do not notice if that is being explained in that forum post.

**Comment by ivan-mogilko** [comment]
The full picture is this:

There is a number of Character properties that correspond to different view values:
* View - this is a active view, whatever it is. It is readonly, which makes total sense, because the current view is chosen according to some logic and factors, and thus should not be set directly.
* NormalView - this is a normal / walking view. It is readonly, and I think that's a mistake.
* SpeechView
* BlinkView
* ThinkView
* IdleView
* more may be added if necessary for the game behavior.

I think these all are logical, except NormalView which is readonly for some reason.

Then, there are following 3 functions that affect view selection:
* ChangeView - sets the new NormalView, and also cancels any ongoing animation.
* LockView - locks character to an arbitrary view, for animation or other purposes.
* UnlockView- unlocks arbitrary view and lets character follow its normal logic.

Here Lock/UnlockView are fine and logical.
ChangeView is strange, because its purpose is not only to set NormalView, but also to switch to it immediately, regardless of current character state: idle animation, or custom animation, or speech animation, etc. This is quite strange, in my opinion.
Also, there's possibly a problem that it does not work fully correctly in that regard, because for some reason it only "unlocks" custom view if there's a idle animation playing, but not a custom animation. Also, I suppose, it may not stop other animations, such as speech animation, correctly, leaving some parameters in the old state (well, I am not fully certain about all the details there).

So, my thinking was:
* if you want to simply assign new normal view but nothing else, then NormalView should be settable (but it's not!).
* if you want to stop custom animation and switch to the default behavior, then UnlockView should be called. This works fine.

What's the place for ChangeView here?
If its purpose is to combine assignment of a new normal view AND stopping current custom anim, then it should be fixed for that. But also it should be decided whether it is allowed to override animation during special states, such as speech, and any others. And if not, then it should only assign normal view, and nothing else.

In other words, my suggestion would be to:
1. Make NormalView settable;
2. Clarify ChangeView's purpose, and fix it to match this purpose, with no undefined or obscure behavior.
3. Decide if there's a need to add more functions to fill any gaps in controlling character views.
